share/availability/test/corrupt_data.go

```diff
@@ -0,0 +1,130 @@
+package availability_test
+
+import (
+	"context"
+	"fmt"
+	mrand "math/rand"
+	"testing"
+
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	ds "github.com/ipfs/go-datastore"
+	dssync "github.com/ipfs/go-datastore/sync"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+)
+
+var _ blockstore.Blockstore = (*MockBlockstore)(nil)
+
+// CorruptBlock is a block where the cid doesn't match the data.
+// It fulfills the blocks.Block interface.
+type CorruptBlock struct {
+	cid  cid.Cid
+	data []byte
+}
+
+func (b *CorruptBlock) RawData() []byte {
+	return b.data
+}
+
+func (b *CorruptBlock) Cid() cid.Cid {
+	return b.cid
+}
+
+func (b *CorruptBlock) String() string {
+	return fmt.Sprintf("[Block %s]", b.Cid())
+}
+
+func (b *CorruptBlock) Loggable() map[string]interface{} {
+	return map[string]interface{}{
+		"block": b.Cid().String(),
+	}
+}
+
+func NewCorruptBlock(data []byte, fakeCID cid.Cid) *CorruptBlock {
+	return &CorruptBlock{
+		fakeCID,
+		data,
+	}
+}
+
+// MockBlockstore is a mock blockstore.Blockstore that saves both corrupted and original data for every block it
```

Comment:
line wrap

Reply:
This one is `gq`'ed, but a different one was not